### DISCUSSION TITLE
FR: Support for Smartcard Authentication with PIN

### DISCUSSION THREAD

**Discussion by tonyyo11** [SAP/macOS-enterprise-privileges#155] [Ideas]
Currently with Privileges 2.0, the `RequireAuthentication` key supports password authentication. If using `AllowCLIBiometricAuthentication`, then Privileges supports Touch ID, if available.

I would like to see, if possible, if smartcard authentication is supported so that users can authenticate with a paired smartcard and PIN. In some regulated organizations, or organizations who are making use of Platform Single Sign On with Smartcard authentication, the user's local password may have been randomly generated at account creation due to authenticating with a smart card or certificate based authentication. Smartcard Authentication may be enforced upon macOS devices so much so that user's may not be aware that there is a local password at all. 

I am not sure how much heavy lifting it would take to support PINs/smartcards. But I would be greatly appreciative if the application did :)

**Comment by mthielemann** [comment]
Created issue #156

**Comment by mthielemann** [comment]
@tonyyo11 Please test with [Privileges 2.2 beta](https://github.com/SAP/macOS-enterprise-privileges/discussions/178).
